backend-services/voting-app/src/main/java/org/cardano/foundation/voting/client/KeriVerificationClient.java

```diff
@@ -145,8 +148,91 @@ public Either<Problem, String> getOOBI(String oobi, Integer maxAttempts) {
         return Either.left(Problem.builder()
                 .withTitle("OOBI_NOT_FOUND")
                 .withDetail("The OOBI was not found after " + attempts + " attempts.")
-                .withStatus(new HttpStatusAdapter(BAD_REQUEST))
+                .withStatus(new HttpStatusAdapter(NOT_FOUND))
                 .build());
     }
 
+    public Either<Problem, Boolean> updateAndVerifyKeyState(String aid, Integer maxAttempts) {
+        val updateUrl = String.format("%s/keystate", keriVerifierBaseUrl);
+        val verifyUrl = String.format("%s/keystate/%s", keriVerifierBaseUrl, aid);
+
+        val headers = new HttpHeaders();
+        headers.add("Content-Type", "application/json");
+
+        val requestBody = new HashMap<String, String>();
+        requestBody.put("pre", aid);
+
+        val entity = new HttpEntity<Map<String, String>>(requestBody, headers);
+
+        // Attempt to update the key state
+        try {
+            val response = restTemplate.exchange(updateUrl, POST, entity, String.class);
+            if (response.getStatusCode().is2xxSuccessful()) {
+                log.info("Key state updated successfully for aid: {}", aid);
+                return verifyKeyState(verifyUrl, maxAttempts);
+            } else {
+                return Either.left(Problem.builder()
+                        .withTitle("KEY_STATE_UPDATE_FAILED")
+                        .withDetail("Failed to update key state.")
+                        .withStatus(new HttpStatusAdapter(response.getStatusCode()))
+                        .build());
+            }
+        } catch (HttpClientErrorException e) {
+            log.error("Unable to update key state, reason: {}", e.getMessage());
+            return Either.left(Problem.builder()
+                    .withTitle("KEY_STATE_UPDATE_ERROR")
+                    .withDetail("Unable to update key state, reason: " + e.getMessage())
+                    .withStatus(new HttpStatusAdapter(e.getStatusCode()))
+                    .build());
+        }
+    }
+
+    private Either<Problem, Boolean> verifyKeyState(String url, Integer maxAttempts) {
+        val headers = new HttpHeaders();
+        headers.add("Content-Type", "application/json");
+        val entity = new HttpEntity<Void>(headers);
+
+        int attempts = (maxAttempts == null) ? 1 : maxAttempts;
+        int attempt = 0;
+
+        while (attempt < attempts) {
+            try {
+                val response = restTemplate.exchange(url, GET, entity, String.class);
+                if (response.getStatusCode().is2xxSuccessful()) {
```

Comment:
if its not complete yet it will still be 200, the response will be `complete: false`

Reply:
https://github.com/cardano-foundation/cf-cardano-ballot/pull/572/commits/2961c6eb7ab091281aabfca2a52513f34223916c

Reply:
Verified this with the help of Jorge - key rotations work